utilities.js

```diff
@@ -0,0 +1,148 @@
+module.exports = isEqual
+
+//In order to sort multi property objects, we need the keys in the same order, it doesn't matter what order they
+//are in, only that they are in a conistent order. This method will also sort child objects
+function sortObjectKeys(object) {
+    var sortedMap = new Map(),
+        keys = Object.keys(object)
+
+    // sort all keys
+    keys = keys.sort((a, b) => a.localeCompare(b))
+
+
+    keys.forEach(function(key) {
+        // handle complex json objects (not likely to exist for SQL table)
+        let exists = object[key]
+        let isObj = typeof object[key] == 'object'
+        let isNotArr = !(object[key] instanceof Array)
+        let isActuallyObject = isObj && isNotArr // all arrays are typeof object, but we want to rule out that this is an array
+
+        if (exists && isActuallyObject) {
+            // if we got an obj, recurse back through it
+            sortedMap[key] = sortObjectKeys(object[key]);
+
+        } else {
+            // everything else, just assign value immediately
+            sortedMap[key] = object[key];
+        }
+    });
+
+    return sortedMap;
+}
+
+//used to sort multi property objects, the result of this sort can change based on the order of keys in
+//object, its purpose is to provide a sort that is the same across objects that have the same keys in the same order
+function compareMultiPropObj(varA, varB) {
+    //for each property in object a, if it is less than the same property in var b return -1 if more return 1
+    for (var prop in varA) {
+        var valFromA = varA[prop]
+        var valFromB = varB[prop]
+        if (valFromA > valFromB) {
+            return 1;
+        } else if (valFromA < valFromB) {
+            return -1;
+        }
```

Comment:
if typeof val == string then use localCompare

doesn't have to be good sort - just has to be deterministic